### PR TITLE
[jnigen] Update links and paths to use dart-lang/native

### PR DESCRIPTION
**Pull request by HosseinYousefi**
Final (?) step to merge jnigen here.

### CODE REVIEW

File: pkgs/jni/CHANGELOG.md [hunk 1]
```diff
@@ -23,11 +23,11 @@
 - Fixed a bug where reading non-null terminated strings would overflow.
 
 ## 0.7.1
-- Removed macOS Flutter plugin until package:jni supports it ([#41](https://github.com/dart-lang/jnigen/issues/41)).
+- Removed macOS Flutter plugin until package:jni supports it ([#780](https://github.com/dart-lang/native/issues/780)).
```
Comment (dcharkes) [inline pkgs/jni/CHANGELOG.md:26]: The old links are forwarded, so it's not needed, but since you've already migrated it, we can keep it.

**Review by dcharkes — APPROVED** [review]
lgtm! Thanks @HosseinYousefi!

File: pkgs/jnigen/example/README.md [hunk 2]
```diff
@@ -4,10 +4,10 @@ This directory contains examples on how to use jnigen.
...
-| [pdfbox_plugin](https://github.com/dart-lang/jnigen/tree/main/jnigen/example/pdfbox_plugin/)             | Example of a flutter plugin which provides bindings to Apache PDFBox library. Currently works on Flutter desktop and Dart standalone on linux. |
-| [notification_plugin](https://github.com/dart-lang/jnigen/tree/main/jnigen/example/notification_plugin/) | Example of a reusable Flutter plugin with custom Java code which uses Android libraries.                                                       |
-| [kotlin_plugin](https://github.com/dart-lang/jnigen/tree/main/jnigen/example/kotlin_plugin)             | Example of using jnigen to generate bindings for Kotlin.                                                                                       |
+| [in_app_java](https://github.com/dart-lang/native/tree/main/pkgs/jnigen/example/in_app_java/)                 | Demonstrates how to include custom Java code in Flutter application and call that using jnigen                                                 |
```
Comment (dcharkes) [inline pkgs/jnigen/example/README.md:7]: We also need to bump the pubspec to release a new version to update the examples tab on pub.dev

Are we ready to release 0.8 ?

File: pkgs/jni/CHANGELOG.md (hunk 1 above)
Comment (HosseinYousefi) [reply pkgs/jni/CHANGELOG.md:26]: That's how I changed them even!

**Pull request merged**